src/ucp/rma/amo_sw.c

```diff
@@ -25,7 +25,8 @@ static size_t ucp_amo_sw_pack(void *dest, void *arg, uint8_t fetch)
 
     atomich->address    = req->send.rma.remote_addr;
     atomich->req.ep_id  = ucp_ep_remote_id(ep);
-    atomich->req.reqptr = fetch ? (uintptr_t)req : 0;
+    atomich->req.req_id = fetch ? ucp_send_request_get_id(req) :
```

Comment:
in some places, maybe use add ucp_request_get_id(req, ep) - when we already have the ep handle (and for receive requests)

Reply:
IMHO it will add a mess how to initialize req_id/ep_id correctly